internal/engine/image_build_and_deployer.go

```diff
@@ -321,25 +323,35 @@ func (ibd *ImageBuildAndDeployer) createEntitiesToDeploy(ctx context.Context,
 					var sidecarInjected bool
 					e, sidecarInjected, err = sidecar.InjectSyncletSidecar(e, injectedRefSelector, ibd.syncletContainer)
 					if err != nil {
-						return nil, err
+						return nil, nil, err
 					}
 					if !sidecarInjected {
-						return nil, fmt.Errorf("Could not inject synclet: %v", e)
+						return nil, nil, fmt.Errorf("Could not inject synclet: %v", e)
 					}
 					injectedSynclet = true
 				}
 			}
 		}
+
+		// This needs to be after all the other injections, to ensure the hash includes the Tilt-generated
+		// image tag, etc
+		e, hashes, err := k8s.InjectPodTemplateSpecHash(e)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "injecting pod template hash")
+		}
+
+		podTemplateSpecHashes = append(podTemplateSpecHashes, hashes...)
+
 		newK8sEntities = append(newK8sEntities, e)
 	}
 
 	for _, depID := range depIDs {
 		if !injectedDepIDs[depID] {
-			return nil, fmt.Errorf("Docker image missing from yaml: %s", depID)
+			return nil, nil, fmt.Errorf("Docker image missing from yaml: %s", depID)
 		}
 	}
 
-	return newK8sEntities, nil
+	return newK8sEntities, podTemplateSpecHashes, nil
```

Comment:
seems kind of weird to return the podTemplateSpecHashes separately, given that they're already in the k8s entities